community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachine.java

```diff
@@ -196,7 +199,16 @@ State run( MutableTransactionState ctx, SPI spi, String statement,
                     {
                         if ( BEGIN.matcher( statement ).matches() )
                         {
-                            ctx.currentTransaction = spi.beginTransaction( ctx.securityContext );
+                            AnyValue txTimeoutMills = params.get( TX_TIMEOUT_MILLIS );
+                            if ( txTimeoutMills instanceof LongValue )
```

Comment:
I think it is better to do `instanceof NumberValue` and then use `NumberValue#longValue()` method. JS only has a single `Number` type which we pack as float.